.bettercodehub.yml

```diff
@@ -0,0 +1,19 @@
+component_depth: 1
+languages:
+- name: java
+  production:
+    include:
+      - server/src/main/*.\.java
+  test:
+    include:
+      - server/src/test/*.\.java
+- name: typescript
+  production:
+    include:
+      - client/src/app/*.\.ts 
+    exclude:
+      - client/src/app/*.\.spec\.ts
+  test:
+    include:
+      - client/src/app/*.\.spec\.ts
+      - client/e2e/*.\.ts
```

Comment:
I think you meant to have `.*` rather than `*.` in these, eg `client/src/app/.*\.ts` rather than `client/src/app/*.\.ts`

Also I think we should include the entire `src` directory for the client since there are some extra things that aren't in `src/app` like environments.

Reply:
You are completely correct about the `.*` – good catch!

I guess I wanted BetterCodeHub to ignore all the config files. The students don't usually change those much, if at all, and some of them are long and kind of unstructured (like the bash deployment setup script). If those files flag things, it's not like we're expecting students to fix them.

That said, if there are specific TypeScript (or JavaScript) files that we do expect the students to modify/maintain we certainly should add them to the list.

Reply:
We have things in `src/environment` and `src/testing`.

Reply:
True, but most of that is essentially config that will be generally left alone by the students.

`src/testing/user.service.mock.ts` looks like the only thing that has any "interesting" content (from a code quality perspective) that students are likely to change.

I think my inclination would be to include `src/testing`, but leave `src/environments` out of it?

Reply:
I feel like the formatting of stuff in `environments` matters too. It'll likely only just be some config variables but that's still typescript that's used by the rest of the app.

Reply:
I'm not sure that either of the analysis tools we're using are likely to catch anything in those, TBH, although thinks like TSLint and `ng lint` might. That said, I see your point and there's really no reason _not_ to include them.